### PR TITLE
sftp filename

### PR DESCRIPTION
**Pull request by peggy-ntt**

### CODE REVIEW

File: jrcc-access-spring-boot-autoconfigure/src/main/java/ca/bc/gov/open/jrccaccess/autoconfigure/plugins/sftp/SftpDocumentInput.java [hunk 1]
```diff
@@ -33,20 +35,20 @@ public void handleMessage(Message<InputStream> message) throws MessagingExceptio
...
 
-
         try {
             logger.debug("Attempting to read downloaded file.");
             String content = getContent(message);
-            logger.info("Successfully red downloaded file.");
-
+            logger.info("Successfully read downloaded file.");
+            logger.debug("Attempting to get file name");
+            String fileName = getFilename(message);
```
Comment (alexjoybc) [inline jrcc-access-spring-boot-autoconfigure/src/main/java/ca/bc/gov/open/jrccaccess/autoconfigure/plugins/sftp/SftpDocumentInput.java]: nice

File: jrcc-access-spring-boot-autoconfigure/src/main/java/ca/bc/gov/open/jrccaccess/autoconfigure/plugins/sftp/SftpDocumentInput.java [hunk 2]
```diff
@@ -67,4 +69,14 @@ private String getContent(Message<InputStream> message) throws IOException {
         return stringBuilder.toString();
     }
 
+    private String getFilename(Message<InputStream> message) throws IOException{
+        MessageHeaders messageHeaders = message.getHeaders();
+        Object filenameObj = messageHeaders.get("file_remoteFile");
+        if(filenameObj == null ){
+            IOException exception = new IOException("corrupted SFTP header. Filename is required.");
```
Comment (alexjoybc) [inline jrcc-access-spring-boot-autoconfigure/src/main/java/ca/bc/gov/open/jrccaccess/autoconfigure/plugins/sftp/SftpDocumentInput.java]: I think it's right to throw an exception here, but no IOexception, IOexception are when you interact with a file, here there is a missing keyvalue in the message header, I would create a custom exception FileNameMissingException that extends DocumentException ... what do you think

Comment (peggy-ntt) [reply jrcc-access-spring-boot-autoconfigure/src/main/java/ca/bc/gov/open/jrccaccess/autoconfigure/plugins/sftp/SftpDocumentInput.java]: ok. I will do that.

File: jrcc-access-spring-boot-autoconfigure/src/main/java/ca/bc/gov/open/jrccaccess/autoconfigure/plugins/sftp/SftpDocumentInput.java [hunk 3]
```diff
@@ -69,14 +74,14 @@ private String getContent(Message<InputStream> message) throws IOException {
...
     }
 
-    private String getFilename(Message<InputStream> message) throws IOException{
+    private String getFilename(Message<InputStream> message) throws DocumentFilenameMissingException{
         MessageHeaders messageHeaders = message.getHeaders();
         Object filenameObj = messageHeaders.get("file_remoteFile");
-        if(filenameObj == null ){
-            IOException exception = new IOException("corrupted SFTP header. Filename is required.");
+        if (!(filenameObj instanceof String )){
+            DocumentFilenameMissingException exception = new DocumentFilenameMissingException("corrupted SFTP header. Filename is required.");
```
Comment (alexjoybc) [inline jrcc-access-spring-boot-autoconfigure/src/main/java/ca/bc/gov/open/jrccaccess/autoconfigure/plugins/sftp/SftpDocumentInput.java]: just throw the exception, no need to have a variable

Comment (alexjoybc) [reply jrcc-access-spring-boot-autoconfigure/src/main/java/ca/bc/gov/open/jrccaccess/autoconfigure/plugins/sftp/SftpDocumentInput.java]: nice to see a custom exception here

**Pull request merged**